### PR TITLE
complete config file documentation

### LINKED ISSUE #212: Updating docs with all config options
It might make sense to update [the config file docs](https://lightning-pose.readthedocs.io/en/latest/source/user_guide/config_file.html) at some point with all the missing options that were added recently.

I always liked[ this page from suite2p](https://suite2p.readthedocs.io/en/latest/settings.html) that listed all possible settings so people are aware of all possible options.

Maybe we could start a list with the things that are missing:

- `training.uniform_heatmaps_for_nan_keypoints`
- `colormap`

### PR DESCRIPTION
**Pull request by themattinthehatt**
The previous config file documentation only included a subset of all possible parameters. These updates include all remaining parameters and also specifies the datatype and default values.

Closes #212

### CODE REVIEW

File: docs/source/user_guide/config_file.rst [hunk 1]
```diff
@@ -24,69 +24,143 @@ The config file contains several sections:
...
+* ``training.train_frames`` (*float or int, default: 1*): this parameter determines how many of the
+  frames assigned to to training data (using ``train_prob``) are actually used for training.
+  This option is generally more useful for testing new algorithms rather than training production
+  models.
+  If the value is a float between 0 and 1 then it is interpreted as the fraction of total train frames.
+  If the value is an integer greater than 1 then it is interpreted as the number of total train frames.
+
+.. _config_num_gpus:
+* ``training.num_gpus`` (*int, default: 1*): the number of GPUs for
+  :ref:`multi-GPU training <multi_gpu_training>`
```
Comment (ksikka) [inline docs/source/user_guide/config_file.rst:99]: Thanks for fixing my broken link.

File: docs/source/user_guide/config_file.rst [hunk 2]
```diff
@@ -24,69 +24,143 @@ The config file contains several sections:
...
+
+* ``training.lr_scheduler`` (*str, default: multisteplr*): reduce the learning rate by a certain
+  factor after a given number of epochs (see ``training.lr_scheduler_params.multisteplr`` below)
+
+* ``training.lr_scheduler_params.multistep_lr``: milestones: epochs at which to reduce learning
+  rate; gamma: factor by which to multiply learning rate at each milestone
+
+* ``training.uniform_heatmaps_for_nan_keypoints`` (*bool, default: true*): how to treat missing
+  hand labels; false to drop, true to force uniform heatmaps. True will lead to better confidence
+  values, while false allows for incompletely labeled data.
```
Comment (ksikka) [inline docs/source/user_guide/config_file.rst:149]: I'm still unclear on why you'd want to use false

Comment (themattinthehatt) [reply docs/source/user_guide/config_file.rst:149]: yes this is a great idea - mind writing this down in the v2.0 notes?

Comment (themattinthehatt) [reply docs/source/user_guide/config_file.rst]: also I tried to clarify the nan handling explanation a bit, and when you might want to use false

Comment (ksikka) [reply docs/source/user_guide/config_file.rst]: Added!

**Comment by ksikka** [comment]
LGTM to merge

**Pull request merged**